clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureBaseClient.groovy

```diff
@@ -46,4 +57,53 @@ public abstract class AzureBaseClient {
       throw new RuntimeException("Failed to create Azure Resource Management Service", e)
     }
   }
+
+  public String getAzureRESTJson(AzureCredentials creds, String baseUrl, String targetUrl, List<String> queryParameters){
+    String result = null
+    String url = AzureUtilities.getAzureRESTUrl(subscriptionId, baseUrl, targetUrl, queryParameters)
+
+    try {
+      // create HTTP Client
+      def config = buildConfiguration(creds)
+      HttpClientBuilder httpClientBuilder = config.create(HttpClientBuilder.class)
```

Comment:
This is a temporary code and only added to work around current limitation of the Azure Java SDK. The source code behind it is pretty much exactly the same with what is currently executed when we create the respective Azure clients (resource manager, network and compute). 
The HTTP client builder requires the proper Azure credentials to construct a proper token; the token can expire so the safest thing is to attempt to get a new one every time (though the same token might be returned if it has not expired).
The method is general enough and should be able to handle any REST calls into Azure and return the JSON string corresponding to the call; this string must be converted to the proper object depending on what REST call is made. At the moment there's only one object that we are interested in and that is the deployment operation object in order to extract the proper error message. Without this the only viable option to understand why a deployment failed is to go into the Azure Portal.
I'm expecting to retouch this code the moment we migrate to the new SDK, hopefully remove it.